service/service_construction.go

```diff
@@ -772,24 +738,27 @@ func generateErc20TransferData(toAddress string, value *big.Int) []byte {
 	return data
 }
 
+// Ref: https://goethereumbook.org/en/transfer-tokens/#forming-the-data-field
 func parseErc20TransferData(data []byte) (*ethcommon.Address, *big.Int, error) {
-	if len(data) != genericTransferBytesLength {
+	if len(data) != 68 {
 		return nil, nil, fmt.Errorf("incorrect length for data array")
 	}
-	methodID := getTransferMethodID()
-	if hexutil.Encode(data[:4]) != hexutil.Encode(methodID) {
+
+	methodBytes, addrBytes, amtBytes := data[:4], data[5:36], data[37:]
```

Comment:
:smile:

Reply:
I think we can use the abi package to make this more robust - will try in a follow-up PR